client/src/modules/cohort/index.jsx

```diff
@@ -1,34 +1,89 @@
 import React, { PureComponent, Fragment } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter, Link } from 'react-router-dom';
+import _map from 'lodash/map';
+import _isEmpty from 'lodash/isEmpty';
 
+import CardItem from 'common/components/CardItem';
+import MessageBox from 'common/components/MessageBox';
 import Toolbar from 'common/components/Toolbar';
+import { getListsForCurrentCohort } from 'modules/shopping-list/model/selectors';
+import { fetchListMetaDataForCurrentCohort } from 'modules/shopping-list/model/actions';
+import { CohortIcon } from 'assets/images/icons';
+import { getCohortDetails } from './model/selectors';
+import { MessageType } from 'common/constants/enums';
 
 class Cohort extends PureComponent {
-  render() {
+  componentDidMount() {
     const {
+      fetchListMetaDataForCurrentCohort,
       match: {
         params: { id }
       }
     } = this.props;
 
+    fetchListMetaDataForCurrentCohort(id);
+  }
+
+  render() {
+    const { cohortDetails, lists } = this.props;
+    const description = cohortDetails && cohortDetails.description;
+    const name = cohortDetails && cohortDetails.name;
```

Comment:
instead of checking every field try following:
```
const { cohortDetails, lists } = this.props;

if (!cohortDetails) {
  return null;
}

const { name, description } = cohortDetails;
```